src/lib/desmo-module.ts

```diff
@@ -87,41 +84,48 @@ export class Desmo {
   }
 
   private async fetchAppOrder(appAddress: string): Promise<AppOrder> {
+    if (this.iexec === undefined) {
+      throw new Error('A connection to iExec is required!');
+    }
     const { orders: appOrders } = await this.iexec.orderbook.fetchAppOrderbook(
       appAddress,
     );
 
-    const appOrder = appOrders && appOrders[0] && appOrders[0].order;
-
-    if (!appOrder) {
-      throw Error(`no apporder found for app ${appAddress}`);
+    if (appOrders.length <= 0) {
```

Comment:
```suggestion
    if (appOrders.length < 1) {
```
It is logically the same but the `<` is easier to read than `<=`